lua/bufferline/buffer.lua

```diff
@@ -2,44 +2,33 @@
 -- buffer.lua
 --
 
-
-local vim = vim
+local concat = table.concat
+local list_slice = vim.list_slice
+local substring = string.sub
 local utils = require'bufferline.utils'
-local split = vim.split
-local join = table.concat
-local len = utils.len
-local slice = utils.slice
-local basename = utils.basename
-local bufname = vim.fn.bufname
-local bufwinnr = vim.fn.bufwinnr
-local matchlist = vim.fn.matchlist
-local nvim_buf_get_option = vim.api.nvim_buf_get_option
-local nvim_get_current_buf = vim.api.nvim_get_current_buf
```

Comment:
I'm a bit confused by all the import changes. Sometimes, this PR adds the `vim.fn` (or other) indirection, sometimes it doesn't. I prefer if we always use the import-into-a-local-then-use pattern. No need to go crazy and remove all `vim.api` references, but if it's already in this form let's keep it that way.

Reply:
I tried to import commonly used ones, and leave out the ones which are not used often / only once. I can just blanket import them though if that is preferred.

Reply:
Done in 23674fdfaeb9868ab9581d64976be941137238e6